packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts

```diff
@@ -67,7 +71,18 @@ export interface FoldedResultListProps {
 /**
  * The `FoldedResultList` headless controller re-organizes results into hierarchical collections (a.k.a. threads).
  */
-export interface FoldedResultList extends ResultList {
+export interface FoldedResultList extends Controller {
+  /**
+   * Using the same parameters as the last successful query, fetch another batch of results, if available.
+   * Particularly useful for infinite scrolling, for example.
+   *
+   * This method is not compatible with the `Pager` controller.
+   */
+  fetchMoreResults(): void;
```

Comment:
One flaw with the division from https://github.com/coveo/ui-kit/commit/114ef0a4303e02143939bd2899187a2628a995cb is that we can no longer extend `ResultList` and `ResultListState`.